desktop/ui/src/main/java/org/datacleaner/windows/DynamoDbDatastoreDialog.java

```diff
@@ -0,0 +1,154 @@
+/**
+ * DataCleaner (community edition)
+ * Copyright (C) 2014 Free Software Foundation, Inc.
+ *
+ * This copyrighted material is made available to anyone wishing to use, modify,
+ * copy, or redistribute it subject to the terms and conditions of the GNU
+ * Lesser General Public License, as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this distribution; if not, write to:
+ * Free Software Foundation, Inc.
+ * 51 Franklin Street, Fifth Floor
+ * Boston, MA  02110-1301  USA
+ */
+package org.datacleaner.windows;
+
+import java.util.List;
+import java.util.Map.Entry;
+
+import javax.inject.Inject;
+import javax.swing.JComponent;
+import javax.swing.JPasswordField;
+import javax.swing.event.DocumentEvent;
+
+import org.apache.metamodel.schema.Schema;
+import org.apache.metamodel.util.SimpleTableDef;
+import org.datacleaner.bootstrap.WindowContext;
+import org.datacleaner.connection.DynamoDbDatastore;
+import org.datacleaner.connection.UpdateableDatastoreConnection;
+import org.datacleaner.guice.Nullable;
+import org.datacleaner.user.MutableDatastoreCatalog;
+import org.datacleaner.user.UserPreferences;
+import org.datacleaner.util.DCDocumentListener;
+import org.datacleaner.util.IconUtils;
+import org.datacleaner.util.ImmutableEntry;
+import org.datacleaner.util.SchemaFactory;
+import org.datacleaner.util.StringUtils;
+import org.datacleaner.util.WidgetFactory;
+import org.datacleaner.widgets.DCComboBox;
+import org.datacleaner.widgets.TableDefinitionOptionSelectionPanel;
+import org.jdesktop.swingx.JXTextField;
+
+import com.amazonaws.regions.Regions;
+
+public class DynamoDbDatastoreDialog extends AbstractDatastoreDialog<DynamoDbDatastore> implements SchemaFactory {
+
+    private static final long serialVersionUID = 1L;
+
+    private final DCComboBox<Regions> _regionField;
+    private final JXTextField _accessKeyField;
+    private final JPasswordField _accessSecretField;
+    private final TableDefinitionOptionSelectionPanel _tableDefinitionWidget;
+
+    @Inject
+    public DynamoDbDatastoreDialog(final WindowContext windowContext, final MutableDatastoreCatalog catalog,
+            @Nullable final DynamoDbDatastore originalDatastore, final UserPreferences userPreferences) {
+        super(originalDatastore, catalog, windowContext, userPreferences);
+
+        _regionField = new DCComboBox<>(Regions.values());
```

Comment:
Didn't we nuke that silly _ with checkstyle, or do I remember wrong?

Reply:
I am not aware that we nuked it, and I largely created this Dialog class by copy-pasting another dialog and fitting it to this type of datastore. The one I copied from had the underscores so I gathered that it was still a used convention :-)

Reply:
I probably remember wrongly, then :-)